apps/src/gamelab/GameLabP5.js

```diff
@@ -38,6 +38,19 @@ var GameLabP5 = function () {
       this.p5.frameRate(this.prevFrameRate || defaultFrameRate);
     }
   };
+
+  this.setLoop = (shouldLoop) => {
+    if (!this.p5) {
+      return;
+    }
+    if (shouldLoop) {
+      // Calling p5.loop() invokes p5.draw(), but we might still be waiting for
+      // animations to load.
+      this.p5._loop = true;
+    } else {
+      this.p5.noLoop();
+    }
```

Comment:
Not sure I'm following what's going on here — why is `setLoop` needed?

Reply:
This is just to encapsulate the logic of putting p5 into loop/noLoop mode when we switch between running setup code and running the full program, since it's not quite as straightforward as calling `loop()` and `noLoop()`.